packages/apps/human-app/frontend/src/modules/worker/oracle-registration/hooks/use-oracle-registration-flow.tsx

```diff
@@ -0,0 +1,49 @@
+import { useParams } from 'react-router-dom';
+import { useCallback, useEffect } from 'react';
+import { useGetOracles } from '@/modules/worker/services/oracles';
+import { useOracleInstructions } from './use-oracle-instructions';
+import { useOracleNavigation } from './use-oracle-navigation';
+import { useOracleRegistration } from './use-oracle-registration';
+
+export function useOracleRegistrationFlow() {
+  const { address: oracleAddress } = useParams<{ address: string }>();
+  const { data: oraclesData } = useGetOracles();
+  const oracleData = oraclesData?.find(
+    (oracle) => oracle.address === oracleAddress
+  );
+
+  const { navigateToJobs, navigateToDiscovery } =
+    useOracleNavigation(oracleAddress);
+  const {
+    handleRegistration,
+    isRegistrationPending,
+    registrationError,
+    isAlreadyRegistered,
+  } = useOracleRegistration(oracleAddress);
+  const { hasViewedInstructions, handleInstructionsView } =
+    useOracleInstructions();
+
+  useEffect(() => {
```

Comment:
Let's move these parts with navigation back to the `page` component.

Right now when you check `registration.page.tsx` code - you see only registration form there and it creates wrong perception that registration form and logic are always there

Would be nice to have it explicit there that if no oracle data or already registered - we navigate away (`isAlreadyRegistered` can be moved out of `useOracleRegistration` hook). This way `useOracleRegistrationFlow` hook will keep only logic related to the actual registration steps